lib/internal/test_runner/test.js

```diff
@@ -608,6 +608,12 @@ class Test extends AsyncResource {
     if (timeout != null && timeout !== Infinity) {
```

Comment:
I'm not sure off the top of my head what happens if `{ timeout: Infinity }` is passed to the test and the CLI option is set to a number, but it seems like the CLI option would take precedent based on this logic.

Reply:
You are right, if `{ timeout: Infinity }` is passed as an option and CLI option set to a number, the CLI number would take precedent. I didn't know what I was thinking. Thanks for spotting this!

Reply:
I made it into an `else if` could you take another look please? 🙏